src/value_ptr.cpp

```diff
@@ -50,6 +50,13 @@ bool ValuePtr::is_float() {
     return value()->is_float();
 }
 
+bool ValuePtr::is_bignum() {
+    if (m_type == Type::Integer)
```

Comment:
Hmmm, not quite right, Bignum is still an integer. The concept comes from ruby itself where an int is either a Fixnum or Bignum (if too big or too small for a Fixnum - in our case long long (?)). So basically m_type == Type::Integer is actually a condition to be a bignum :)

`ruby
return m_type == Type::Integer && value()->is_integer() && value()->as_integer()->is_bignum();
`

Reply:
Are you sure? `ValuePtr::Type::Integer` means that we use `ValuePtr::m_integer` instead of hydrating a full `IntegerValue` object (that may be either Fixnum or Bignum)

Since `ValuePtr::m_integer` is a `long long`, it seems to me that it must always be a Fixnum. But maybe I'm misunderstanding something?

Reply:
Ah I see, this actually works in ValuePtr, the type I was looking at was ValueType (in which case it'd be integer for both), but when it comes to ValuePtr::Type then Fixnums are Integer and Bignums are Pointer so you are right!